### ISSUE TITLE
Why are m7-8 plots sometimes skipped in IS diagnostics reports?

### ISSUE THREAD

**Issue by chrisbc** [GNS-Science/nzshm-opensha#238]
Chris Rollins noticed that when we have spiky MFD sometimes this will cause M7-8 plot to not work.Q:  Is there any way to fix that?

> [@Chris Rollins](https://nshmrevisionproject.slack.com/team/U01AAQXK2AF) interesting - when I look at the [opensha solution diags report](http://simple-toshi-ui.s3-website-ap-southeast-2.amazonaws.com/InversionSolution/SW52ZXJzaW9uU29sdXRpb246MTAzOTQ4/DiagnosticReportTab) for this that plot is skipped . There must be a reason why Kevin thinks these are not useful ? But I'll raise a ticket as we need to understand why. fyi [@Oakley Jurgens](https://nshmrevisionproject.slack.com/team/U019P6CBH0U)

 - See the [solution report](http://simple-toshi-ui.s3-website-ap-southeast-2.amazonaws.com/InversionSolution/SW52ZXJzaW9uU29sdXRpb246MTAzOTQ4/DiagnosticReportTab)  linked (same as  below ...)
 - also see the [toshiUI view](http://simple-toshi-ui.s3-website-ap-southeast-2.amazonaws.com/GeneralTask/R2VuZXJhbFRhc2s6MTAxODEw
 /ChildTasks?clipBoard=eyJmaWx0ZXIiOnsiZGF0YSI6W3siayI6ImJfYW5kX24iLCJ2IjpbInsndGFnJzogJ2IgPSAxLjA3OSwgTiA9IDIzLjAnLCAnYic6IDEuMDc5LCAnTic6IDIzLjB9Iiwieyd0YWcnOiAnYiA9IDAuOTYxLCBOID0gMTYuNycsICdiJzogMC45NjEsICdOJzogMTYuN30iXX0seyJrIjoic2NhbGluZ19jX3ZhbHMiLCJ2IjpbIjQuMSJdfV19LCJzaG93TGlzdCI6ZmFsc2UsInNob3dGaWx0ZXIiOnRydWUsImdlbmVyYWxWaWV3cyI6WyJTb2x1dGlvbiBNRkRzIChpbmNyZW1lbnRhbCkiLCJTbGlwIFJhdGUgU2NhdHRlciBwbG90IiwiUGFydGljaXBhdGlvbiBNOCB0byBNOSIsIlBhcnRpY2lwYXRpb24gPj0gTTkiLCJQYXJ0aWNpcGF0aW9uIE03IHRvIE04IiwiU2xpcCBSYXRlIChTb2x1dGlvbiAvIFRhcmdldCkiXSwibmFtZWRGYXVsdHNWaWV3IjoiTnVjbGVhdGlvbiBDdW11bGF0aXZlIiwibmFtZWRGYXVsdHNMb2NhdGlvbnMiOlsiQWxwaW5lIEdlb3JnZSB0byBKYWNrc29ucyJdLCJyZWdpb25hbFZpZXdzIjpbIlNvbHV0aW9uIE1GRC1TYW5zIFRWWihpbmNyZW1lbnRhbCkiXSwibm9uUmVnaW9uYWxWaWV3cyI6WyJJbmNyZW1lbnRhbCJdLCJyZXBvcnRUYWIiOjAsInBhcmVudEZhdWx0IjoiQWxwaW5lX0thbmllcmVfdG9fU3ByaW5nc19KdW5jdGlvbiIsInBhcmVudEZhdWx0Vmlld3MiOlsiQ28tcnVwdHVyZSByYXRlcyIsIlNlY3Rpb24gTUZEIChJbmNyZW1lbnRhbCkiXX0%3D)

**Comment by chrisbc** [comment]
FYI @chrisdicaprio

**Comment by voj** [comment]
@chrisbc  I'm assuming this issue is about participation plots and you're wondering why for this particular solution, there are no plots in the 7-8M range.

The input min mag for the solution is 7, and the actual min mag is 7.048983789490542. I'm assuming that's a bucket centre. In `ParticipationRatePlot`, there is this bit of code

```Java
		if (maxMag > 6d && minMag <= 6d) {
			minMags.add(6d);
			magLabels.add("M≥6");
			magPrefixes.add("m6");
		}
		
		if (maxMag > 7d && minMag <= 7d) {
			minMags.add(7d);
			magLabels.add("M≥7");
			magPrefixes.add("m7");
		}
		
		if (maxMag > 8d) {
			minMags.add(8d);
			magLabels.add("M≥8");
			magPrefixes.add("m8");
		}
		
		if (maxMag > 9d) {
			minMags.add(9d);
			magLabels.add("M≥9");
			magPrefixes.add("m9");
		}
```

So 6 and 7 are not used as min mags in plots. I'm assuming this means the plots you are interested in are skipped.

Lots of assumptions, but it looks like our min mag is just too high to create the plots.

**Comment by chrisbc** [comment]
@voj yes you guessed right  - it is the participation plot Chris R is looking for - the one  for magnitudes between 7 and 8 is  skipped for some reason.

**Comment by voj** [comment]
Fixed upstream in c40f37cfdbf5131e357b28b5c1bd39a02d1dba60
